### PR TITLE
Migration to GitHubSdk 0.5

### PR DESCRIPTION
**Pull request by plweegie**
I thought it might be a good idea to connect the app to the new version of the SDK, which is much more flexible and will allow for more features to be added later, following up from https://github.com/Meisolsson/GitHubSdk/pull/13. There are a couple of breaking changes and I have rewritten a number of classes and tests to take that into account. Because classes in `com.github.pockethub.android.ui.item.news` have been the most heavily affected, I also decided to migrate that package to Kotlin.

### CODE REVIEW

File: app/src/main/java/com/github/pockethub/android/ui/issue/IssueDashboardPagerAdapter.java [hunk 1]
```diff
@@ -26,7 +26,15 @@
 import java.util.HashMap;
 import java.util.Map;
 
-import static com.github.pockethub.android.core.issue.IssueFilter.*;
```
Comment (Meisolsson) [inline app/src/main/java/com/github/pockethub/android/ui/issue/IssueDashboardPagerAdapter.java]: Why was this changed?

File: app/src/test/java/com/github/pockethub/android/util/ConvertUtilsTest.java [hunk 2]
```diff
@@ -30,32 +34,54 @@
...
-				.login(REPO_OWNER_LOGIN)
-				.build();
-
-		repo = Repository.builder().
-				name(REPO_NAME)
-				.owner(user)
-				.build();
+
+		event = GitHubEvent.builder()
+                .repo(new GitHubEvent.RepoIdentifier() {
```
Comment (Meisolsson) [inline app/src/test/java/com/github/pockethub/android/util/ConvertUtilsTest.java]: This need to be fixed, I'll get to today. `GitHubEvent.RepoIdentifier` should really have a builder.

There will be a new release today.

Comment (Meisolsson) [reply app/src/test/java/com/github/pockethub/android/util/ConvertUtilsTest.java]: Version `0.5.1` has a builder for this and is released now.

Comment (plweegie) [reply app/src/test/java/com/github/pockethub/android/util/ConvertUtilsTest.java]: Ok great, a builder will help a lot.

File: app/src/main/java/com/github/pockethub/android/ui/issue/IssueDashboardPagerAdapter.java (hunk 1 above)
Comment (plweegie) [reply app/src/main/java/com/github/pockethub/android/ui/issue/IssueDashboardPagerAdapter.java]: This does not look good at all. I will declutter this and any other static imports

**Comment by plweegie** [comment]
Thanks for the review. I will fix the tests for 0.5.1 and optimize some of the other things you mentioned as soon as I can. Will also keep PRs simple in the future.

**Comment by plweegie** [comment]
I made the changes for the minor comments. I have the migration to 0.5.1 with all the tests passing on a feature branch. I'm assuming you'd prefer that as a separate PR, otherwise I can push the changes here. Let me know.

**Comment by Meisolsson** [comment]
Well the changes to 0.5.1 would probably be nice to have in this since it's just for the `RepoIdentifier`. Other than that the only thing i still want changed is the imports which changed all over the place.

**Comment by plweegie** [comment]
Ok, this should do it then.

**Comment by Meisolsson** [comment]
Looks good.
